Entities/Items/Projectiles/Arrow.as

```diff
@@ -421,6 +423,31 @@ void AddArrowLayer(CBlob@ this, CBlob@ hitBlob, CSprite@ sprite, Vec2f worldPoin
 
 		f32 angle = velocity.Angle();
 		arrow.RotateBy(-angle - hitBlob.getAngleDegrees(), Vec2f(0, 0));
+
+		//track time until arrow is destroyed
+
+		//initialize arrays
+		if (!hitBlob.exists("stuck_arrow_names"))
+		{
+			string[] names;
+			hitBlob.set("stuck_arrow_names", names);
+
+			uint[] times;
+			hitBlob.set("stuck_arrow_times", times);
+		}
+
+		//save details of arrow so it can decay
+		hitBlob.push("stuck_arrow_names", arrow.name);
```

Comment:
Server also runs this code, but never runs DecayStuckArrow.as, meaning this array will just keep increasing in size until the blob dies.

Before running AddArrowLayer, do a isServer check since server doesnt need to run anything here